api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/RoutineCalculation.java

```diff
@@ -67,6 +67,7 @@ public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Obj
 		Concept regimeConcept = hivMetadata.getRegimeConcept();
 		Date latestVlLowerDateLimit = EptsCalculationUtils.addMonths(context.getNow(), -12);
 		EncounterType labEncounterType = hivMetadata.getMisauLaboratorioEncounterType();
+		String criteria = (String) params.get("criteria");
```

Comment:
I am NOT sure I understand how to do this here, but I will try it on the cohortDefinition

Reply:
In the cohort definition, the parameters are String, Object pairs `void addCalculationParameter(String name, Object value)`

So you can pass in an ENUM object as the value instead of a string value to differentiate between breastfeeding and adultnChildren.

```
public enum CustomEnum {
    BREASTFEEDINGPREGNANT,
    ADULTCHILDREN
}

# usage
addCalculationParameter("parameterName", CustomEnum.ADULTCHILDREN)
```

Reply:
Yes, I got confusion a bit where to put those, but my initial idea was in the cohortDefinition as I mentioned. So this enumeration can be created in the same class or I need to put somewhere else say the constant class?

Reply:
Like how OpenMRS is doing https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/util/OpenmrsConstants.java#L1289-L1293

Reply:
Yes  `EptsReportConstants` is a good place

Reply:
I will apply those somewhere like this 
cd.addSearch("routine",
		   EptsReportUtils.map(getPatientsWhoAreOnRoutine(String.valueOf(EptsReportConstants.CustomEnum.ADULTCHILDREN)), "onDate=${endDate},location=${location}"));

Reply:
No I was thinking you pass in the ENUM and then only parse it as a string when it gets used so:
`cd.addSearch("routine",
EptsReportUtils.map(getPatientsWhoAreOnRoutine(EptsReportConstants.CustomEnum.ADULTCHILDREN)), "onDate=${endDate},location=${location}"));`

Then in RoutineCalculation use it as follows:
```
CustomEnum criteria = (CustomEnum) params.get("criteria");
...criteria.equals(CustomEnum.ADULTCHILDREN)
```

Reply:
of course don't name the enum `CustomEnum` use something more descriptive :)

Reply:
Done, you can have a look